data/campaigns/Heir_To_The_Throne/scenarios/05b_Isle_of_the_Damned.cfg

```diff
@@ -341,60 +341,58 @@
             message= _ "Surprise! Searching for magi, and all I get is these foul humans!"
         [/message]
     [/event]
+#endif
 #enddef
 
     [event]
         name=prestart
 
-        #
-        # Determine the contents of the temples:
-        # - on easy, Moremirmu shows up in 1 close temple and the others are empty
-        # - on normal, Moremirmu shows up in 1 close temple and the far one has the revenant
-        # - on hard, Moremirmu and the revenant show up in the 2 close temples (far one is empty)
-        #
-        {RANDOM 1..2}
-        [if]
-            [variable]
-                name=random
-                numerical_equals=1
-            [/variable]
-#ifdef EASY
-            [then]
+        {RANDOM 1..6}
+        [switch]
+            variable=random
+            [case]
+                value=6
+
+                {EMPTY_TEMPLE_TRAP 9 15}
+                {XAKAE_TRAP 10 17}
                 {MOREMIRMU_TRAP 11 13}
+            [/case]
+            [case]
+                value=5
+
+                {XAKAE_TRAP 9 15}
                 {EMPTY_TEMPLE_TRAP 10 17}
+                {MOREMIRMU_TRAP 11 13}
+            [/case]
+            [case]
+                value=4
+
                 {EMPTY_TEMPLE_TRAP 9 15}
-            [/then]
-            [else]
                 {MOREMIRMU_TRAP 10 17}
-                {EMPTY_TEMPLE_TRAP 11 13}
-                {EMPTY_TEMPLE_TRAP 9 15}
-            [/else]
-#endif
-#ifdef NORMAL
-            [then]
-                {MOREMIRMU_TRAP 11 13}
-                {EMPTY_TEMPLE_TRAP 10 17}
+                {XAKAE_TRAP 11 13}
+            [/case]
+            [case]
+                value=3
+
                 {XAKAE_TRAP 9 15}
-            [/then]
-            [else]
                 {MOREMIRMU_TRAP 10 17}
                 {EMPTY_TEMPLE_TRAP 11 13}
-                {XAKAE_TRAP 9 15}
-            [/else]
-#endif
-#ifdef HARD
-            [then]
-                {MOREMIRMU_TRAP 11 13}
-                {XAKAE_TRAP 10 17}
-                {EMPTY_TEMPLE_TRAP 9 15}
-            [/then]
-            [else]
-                {MOREMIRMU_TRAP 10 17}
+            [/case]
+            [case]
+                value=2
+
+                {MOREMIRMU_TRAP 9 15}
+                {EMPTY_TEMPLE_TRAP 10 17}
                 {XAKAE_TRAP 11 13}
-                {EMPTY_TEMPLE_TRAP 9 15}
+            [/case]
+            [else]
+                # value=1 and "can't happen, but just in case" case
```

Comment:
Many's the time I've had a switch fail because someone added a value to the range. I'd rather the programmer scratch his head at no-workie than the user complain about a crash.